android-components/components/feature/media/src/test/java/mozilla/components/feature/media/service/MediaSessionServiceDelegateTest.kt

```diff
@@ -460,9 +469,59 @@ class MediaSessionServiceDelegateTest {
         verify(controller).pause()
     }
 
+    @Test
+    fun `WHEN on API level is 31 startForeground throws an exception THEN catch and pass the exception to the crash reporter`() = runTestOnMain {
+        val crashReporter: CrashReporting = mock()
+        val service: AbstractMediaSessionService = mock()
+        val delegate = MediaSessionServiceDelegate(testContext, service, mock(), crashReporter)
+        delegate.onCreate()
+        val notification: Notification = mock()
+        delegate.notificationHelper = coMock {
+            doReturn(notification).`when`(this).create(mock(), delegate.mediaSession)
+        }
+
+        val exception = ForegroundServiceStartNotAllowedException("Test thrown exception")
+        doThrow(exception).`when`(service).startForeground(anyInt(), any())
+
+        setSdkInt(31)
+        delegate.startForeground(mock())
```

Comment:
nit: I think it would make sense to include setting the SDK with the setup part of the test above
```suggestion
        delegate.notificationHelper = coMock {
            doReturn(notification).`when`(this).create(mock(), delegate.mediaSession)
        }

        val exception = ForegroundServiceStartNotAllowedException("Test thrown exception")
        doThrow(exception).`when`(service).startForeground(anyInt(), any())
        setSdkInt(31)

        delegate.startForeground(mock())
```